### ISSUE TITLE
additional_properties does not work for console.list_permitted_lpars

### ISSUE THREAD

**Issue by mariahsarica** [zhmcclient/python-zhmcclient#1410]
**Describe the bug**
`additional_properties` for `Console.list_permitted_lpars()` does not work. 
Perhaps the following block of code may be needed before [line 798 in _console.py](https://github.com/zhmcclient/python-zhmcclient/blob/master/zhmcclient/_console.py#L789) (block of code appears in `list_permitted_adapters` but not in `list_permitted_lpars`)
```python
        if additional_properties:
            ap_parm = 'additional-properties={}'.format(
                ','.join(additional_properties))
            query_parms.append(ap_parm)
```
**To Reproduce**
Here is my python script where the additional properties specified are not part of the output

```python
session = zhmcclient.Session(host, userid, password, verify_cert=verify_cert)
client = zhmcclient.Client(session)
console = client.consoles.console

filter_args = {'cpc-name': 'A89', 'status': 'operating'}
additional_props = ['os-name', 'sysplex-name']

lpars = client.consoles.console.list_permitted_lpars(
    filter_args=filter_args,
    additional_properties=additional_props,
    full_properties=False)

for lpar in lpars:
    print(lpar._properties)
```
output:
```
{'parent': '/api/cpcs/e9d1ac53-4b67-3cc0-bf33-d9a4b3653c18', 'class': 'logical-partition', 'object-id': '06f2eae8-c817-3506-b976-fc301f14f826', 'name': 'R7C', 'activation-mode': 'general', 'status': 'operating', 'has-unacceptable-status': False, 'object-uri': '/api/logical-partitions/06f2eae8-c817-3506-b976-fc301f14f826'}
{'parent': '/api/cpcs/e9d1ac53-4b67-3cc0-bf33-d9a4b3653c18', 'class': 'logical-partition', 'object-id': '943af48d-395d-3562-b50c-e8020e71bd89', 'name': 'S5A', 'activation-mode': 'general', 'status': 'operating', 'has-unacceptable-status': False, 'object-uri': '/api/logical-partitions/943af48d-395d-3562-b50c-e8020e71bd89'}
{'parent': '/api/cpcs/e9d1ac53-4b67-3cc0-bf33-d9a4b3653c18', 'class': 'logical-partition', 'object-id': '4ee42b46-e294-31b4-b359-234bbf64cd26', 'name': 'TA0', 'activation-mode': 'general', 'status': 'operating', 'has-unacceptable-status': False, 'object-uri': '/api/logical-partitions/4ee42b46-e294-31b4-b359-234bbf64cd26'}
```
**Expected behavior**
I'd expect the output to also contain `os-name` and `sysplex-name` properties

**Environment information**

* HMC version: 2.16.0 Bundle H21

* Output of `python -c "import zhmcclient; print(zhmcclient.debuginfo())"`:
```
os_name: Darwin
os_version: 23.2.0
cpu_arch: x86_64
bit_size: 64
unicode_size: wide
python_impl: CPython
python_impl_version: 3.11.6
python_version: 3.11.6
zhmcclient_version: 1.13.1
```

**Comment by andy-maier** [comment]
Hi @mariahsarica 
Your proposed fix was right on. I have incorporated that in the PR.

I also analyzed why the test did not find that issue, and it turns out there was incorrect logic in the unit test. After fixing that, it turned out the processing of the additional-properties in the zhmcclient mock support, and the optimization in the list methods also needed to be fixed. All of that is in PR #1411 .
